lib/factorial.rb

```diff
@@ -1,4 +1,13 @@
 # Computes factorial of the input number and returns it
+
 def factorial(number)
-  raise NotImplementedError
+  raise ArgumentError, "Invalid Number" if number.nil?
+
+  total = 1
+
+  number.times do
```

Comment:
I'd recommend making this one of the following for clarity:
```ruby
(number-1).times do |i|
   total *= (i+1)
end
```

or

```ruby
while number > 1
   total *= number
   number -= 1
end
```

This way you won't have code where the value of number is changing inside a number.times' loop.